dymos/visualization/timeseries_plots.py

```diff
@@ -0,0 +1,193 @@
+
+        # add labels, title, and legend
+        ax.set_xlabel(time_label)
+        ax.set_ylabel(var_label)
+        fig.suptitle(title)
+
+        # Plot each phase
+        for iphase, phase_name in enumerate(phase_names):
+            if phases_node_path:
+                var_name_full = f'{phases_node_path}.{phase_name}.timeseries.states:{var_name}'
+                time_name = f'{phases_node_path}.{phase_name}.timeseries.time'
+            else:
+                var_name_full = f'{phase_name}.timeseries.states:{var_name}'
+                time_name = f'{phase_name}.timeseries.time'
+
+            # Get values
+            var_val = last_solution_case.outputs[var_name_full]
+            time_val = last_solution_case.outputs[time_name]
+
+            # Plot the data
+            color = cm.colors[iphase % 20]
+
+            ax.plot(time_val, var_val, marker='o', linestyle='None', label='solution', color=color)
+
+            # get simulation values, if plotting simulation
+            if simulation_record_file:
+                # if the phases_node_path is empty, need to pre-pend names with "sim_traj."
+                #   as that is pre-pended in Trajectory.simulate code
+                sim_prefix = "" if phases_node_path else "sim_traj."
+                var_val_simulate = last_simulation_case.outputs[sim_prefix + var_name_full]
+                time_val_simulate = last_simulation_case.outputs[sim_prefix + time_name]
+                ax.plot(time_val_simulate, var_val_simulate, linestyle='--', label='simulation',
+                        color=color)
+
+        # Create two legends
+        #   Solution/Simulation legend
+        solution_line = mlines.Line2D([], [], color='black', marker='o', linestyle='None',
+                                      label='Solution')
+        if simulation_record_file:
+            simulation_line = mlines.Line2D([], [], color='black', linestyle='--',
+                                            label='Simulation')
+            sol_sim_legend = plt.legend(handles=[solution_line, simulation_line],
+                                        loc='upper center', bbox_to_anchor=(0., -0.12), shadow=True)
+        else:
+            sol_sim_legend = plt.legend(handles=[solution_line],
+                                        loc='upper center', bbox_to_anchor=(0., -0.12),
+                                        shadow=True)
+        plt.gca().add_artist(sol_sim_legend)
+
+        #   Phases legend
+        handles = []
+        for iphase, phase_name in enumerate(phase_names):
+            patch = mpatches.Patch(color=cm.colors[iphase], label=phase_name)
+            handles.append(patch)
+        plt.legend(handles=handles, loc='upper right', ncol=len(phase_names), shadow=True,
+                   bbox_to_anchor=(1.1, -0.12), title='Phases')
+
+        plt.subplots_adjust(bottom=0.23, top=0.9)
+
+        # save to file
+        plot_file_path = os.path.join(plot_dir_path, f'test_{var_name}.png')
```

Comment:
I wouldn't use `test_` as the file prefix for all cases here.  I'd prefer using `f'{var_name}.png'`.

Reply:
Wow, I have no idea why I had that `test_` in there. Sorry!